lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -2479,12 +2484,14 @@ public function ConditionalPrimary()
         // Peek beyond the matching closing parenthesis ')'
         $peek = $this->peekBeyondClosingParenthesis();
 
-        if (in_array($peek['value'], ["=",  "<", "<=", "<>", ">", ">=", "!="]) ||
-            in_array($peek['type'], [Lexer::T_NOT, Lexer::T_BETWEEN, Lexer::T_LIKE, Lexer::T_IN, Lexer::T_IS, Lexer::T_EXISTS]) ||
-            $this->isMathOperator($peek)) {
-            $condPrimary->simpleConditionalExpression = $this->SimpleConditionalExpression();
+        if ($peek !== null) {
+            if (in_array($peek['value'], ["=",  "<", "<=", "<>", ">", ">=", "!="]) ||
```

Comment:
This if can be combined with the previous one

Reply:
@lcobucci Do you mean that you'd prefer
```php
if ($peek !== null &&
    in_array($peek['value'], ["=",  "<", "<=", "<>", ">", ">=", "!="]) ||
    in_array($peek['type'], [Lexer::T_NOT, Lexer::T_BETWEEN, Lexer::T_LIKE, Lexer::T_IN, Lexer::T_IS, Lexer::T_EXISTS]) ||
    $this->isMathOperator($peek)) {
```

Reply:
The idea is to not have nested ifs and reduce the cyclomatic complexity, so yes that is the end result (probably adding parenthesis).

I would really prefer we used objects instead of arrays for parsing and tokenising things but this is not the right time for this :stuck_out_tongue:

Reply:
Done.